library/alloc/src/boxed.rs

```diff
@@ -1011,6 +1011,8 @@ impl<T: ?Sized> Box<T> {
     /// function is called twice on the same raw pointer.
     ///
     /// The raw pointer must point to a block of memory allocated by the global allocator.
+    /// And you should enforce the aliasing rule by ensuring nothing else uses the
+    /// raw pointer after calling this function.
```

Comment:
Please don't start a sentence with "and", that's not proper writing.  Maybe just copy the wording from `Vec::from_raw_parts`?

Also, it's not true that the raw pointer may never be used again. If you `mem::forget` the `Box`, you can use the raw pointer again. But the same is true for `Vec` so maybe we deliberately don't talk about this in the docs? Cc @rust-lang/opsem